ECR login error on Windows runner

The ECR login command is causing an error on Windows platform runners, resulting in the following message:

```
##[error]Command failed: $(aws ecr get-login --region us-east-1 --no-include-email)
'$' is not recognized as an internal or external command,
operable program or batch file.
```

Logic will need to be added to handle this login command via PowerShell.

--- a/README.md
+++ b/README.md
@@ -92,6 +92,7 @@ with:
 * Create an IAM user with the ability to push to ECR (see [example policies](https://docs.aws.amazon.com/AmazonECR/latest/userguide/ecr_managed_policies.html))
 * Create and download access keys
 * Save `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as secrets in your GitHub repo
+* Ensure the repo you are trying to push to already exists, if not create with `aws ecr create-repository` before pushing
 * Modify sample below and include in your workflow `.github/workflows/*.yml` file
 
 ```yaml

--- a/src/docker.js
+++ b/src/docker.js
@@ -65,6 +65,8 @@ const isEcr = registry => registry && registry.includes('amazonaws');
 
 const getRegion = registry => registry.substring(registry.indexOf('ecr.') + 4, registry.indexOf('.amazonaws'));
 
+const isWindows = () => process.env.RUNNER_OS === 'Windows';
+
 const login = () => {
   const registry = core.getInput('registry', { required: true });
   const username = core.getInput('username');
@@ -74,7 +76,15 @@ const login = () => {
   if (isEcr(registry)) {
     const region = getRegion(registry);
     core.info(`Logging into ECR region ${region}...`);
-    cp.execSync(`$(aws ecr get-login --region ${region} --no-include-email)`);
+
+    // Determine whether to run bash or PowerShell version of login command
+    if (isWindows()) {
+      cp.execSync(
+        `aws ecr get-login-password --region ${region} | docker login --username AWS --password-stdin ${registry}`
+      );
+    } else {
+      cp.execSync(`$(aws ecr get-login --region ${region} --no-include-email)`);
+    }
   } else if (username && password) {
     core.info(`Logging into Docker registry ${registry}...`);
     cp.execSync(`docker login -u ${username} --password-stdin ${registry}`, {

--- a/tests/docker.test.js
+++ b/tests/docker.test.js
@@ -158,6 +158,19 @@ describe('core and cp methods', () => {
       expect(cp.execSync).toHaveBeenCalledWith(`$(aws ecr get-login --region us-east-1 --no-include-email)`);
     });
 
+    test('ECR Windows login', () => {
+      process.env.RUNNER_OS = 'Windows';
+      const registry = '123456789123.dkr.ecr.us-east-1.amazonaws.com';
+
+      core.getInput.mockReturnValueOnce(registry).mockReturnValueOnce('').mockReturnValueOnce('');
+
+      docker.login();
+
+      expect(cp.execSync).toHaveBeenCalledWith(
+        `aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin ${registry}`
+      );
+    });
+
     test("returns undefined if empty login and doesn't execute command", () => {
       core.getInput.mockReturnValueOnce('').mockReturnValueOnce('').mockReturnValueOnce('');
 

